doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1591,21 +1592,172 @@ Numeral notations
+
+            .. coqtop:: all
+
+               Check 0.
+               Check 1.
+               Check 2.
+
+      :n:`via @qualid__ind using ( {+, @qualid @qualid } )`
+         When using this variant, the :n:`@qualid` type above no
+         longer needs to be an inductive. It is mapped to the
+         inductive type (without parameters) :n:`@qualid__ind` according to the provided
+         list of pairs :n:`@qualid @qualid` whose first component is a
+         constant of type :n:`@qualid` above
+         (or a function of type :n:`{* _ -> } @qualid`) and the second a
+         constructor of type :n:`@qualid__ind`. The above type
+         :n:`@qualid` is then replaced by :n:`@qualid__ind` in the
+         above parser and printer types.
+
+         .. note::
+            To use a sort as target type, use an :ref:`abbreviation <Abbreviations>`
+            as in example below.
+
+         The following example parses and prints :g:`prod unit (prod unit unit)` as :g:`3`.
+         The inductive :g:`I` will be used as :n:`@qualid__ind`.
+
+         .. coqtop:: in reset
+
+            Inductive I :=
+            | Iempty : I
+            | Iunit : I
+            | Iprod : I -> I -> I.
+
+            Definition of_uint (x : Numeral.uint) : I :=
+              let fix f n :=
+                match n with
+                | O => Iempty
+                | S O => Iunit
+                | S n => Iprod Iunit (f n)
+                end in
+              f (Nat.of_num_uint x).
+
+            Definition to_uint (x : I) : Numeral.uint :=
+              let fix f i :=
+                match i with
+                | Iempty => O
+                | Iunit => 1
+                | Iprod i1 i2 => f i1 + f i2
+                end in
+              Nat.to_num_uint (f x).
+
+            Inductive prod (A : Set) (B : Set) : Set := pair : A -> B -> prod A B.
+
+         .. coqtop:: in
+
+            Notation nSet := Set (only parsing).
+            Numeral Notation nSet via I using (Empty_set Iempty, unit Iunit, prod Iprod)
+              of_uint to_uint : type_scope.
+
+         .. coqtop:: all
+
+            Local Open Scope type_scope.
+
+            Check Empty_set.
+            Check unit.
+            Check prod unit unit.
+            Check prod unit (prod unit unit).
+
+         .. coqtop:: all
+
+            Set Printing All.
+            Check 0.
+            Check 1.
+            Check 2.
+            Check 3.
+
+         .. note::
+
+            When the terms mapped to the constructors of
+            :n:`@qualid__ind` have implicit arguments, those are
+            ignored when translating to the inductive type
+            :n:`@qualid__ind` before calling :n:`@qualid__parse` and
+            filled with implicit argument holes :g:`_` after
+            :n:`@qualid__print`. Note that this makes numeral
+            notations fragile with respect to implicit argument
+            status. Example:
```

Comment:
```suggestion
            Terms mapped to the constructors of
            :n:`@qualid__ind` that have implicit arguments are
            ignored when translating to the inductive type
            :n:`@qualid__ind` before calling :n:`@qualid__parse` and
            filled with implicit argument holes :g:`_` after
            :n:`@qualid__print`. This makes numeral
            notations fragile in relation to implicit arguments. Example:
```

Reply:
This changes the meaning of the first sentence (the things that are ignored are the arguments, not the terms) but this points out that the sentence is ambiguous. I have to rephrase that.